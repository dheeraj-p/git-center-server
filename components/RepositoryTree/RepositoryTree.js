import { Fragment } from 'react';
import RepositoryTreeItems from '../RepositoryTreeItems/RepositoryTreeItems';

export default function RepositoryTree({
  repository,
  name,
  path,
  branch,
  content,
}) {
  return (
    <Fragment>
      <p className='text-muted'>{path}</p>
      <div className='list-group mt-3'>
        <RepositoryTreeItems
          repository={repository}
          branch={branch}
          content={content}
        />
      </div>
    </Fragment>
  );
}
