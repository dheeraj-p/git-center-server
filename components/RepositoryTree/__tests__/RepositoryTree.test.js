import { shallow } from 'enzyme';
import RepositoryTree from '../RepositoryTree';

describe('RepositoryTree', () => {
  let content;
  beforeEach(() => {
    content = [
      {
        type: 'blob',
        name: 'testBlob',
        path: 'src/blob.js',
      },
      {
        type: 'tree',
        name: 'src',
        path: '/src',
      },
    ];
  });

  it('should render properly', () => {
    const wrapper = shallow(
      <RepositoryTree
        content={content}
        repository='testRepo'
        branch='master'
        path='test/path'
      />
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
