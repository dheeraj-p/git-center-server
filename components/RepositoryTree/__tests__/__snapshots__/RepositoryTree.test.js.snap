// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryTree should render properly 1`] = `
"<Fragment>
  <p className=\\"text-muted\\">
    test/path
  </p>
  <div className=\\"list-group mt-3\\">
    <RepositoryTreeItems repository=\\"testRepo\\" branch=\\"master\\" content={{...}} />
  </div>
</Fragment>"
`;
